extensions/androidx-paging3/src/commonTest/kotlin/app/cash/sqldelight/paging3/KeyedQueryPagingSourceTest.kt

```diff
@@ -37,91 +36,70 @@ class KeyedQueryPagingSourceTest {
 
   private lateinit var driver: SqlDriver
   private lateinit var transacter: Transacter
+  private lateinit var source: KeyedQueryPagingSource<Long, Long>
 
-  @BeforeTest fun before() {
+  private suspend fun before() {
     driver = provideDbDriver()
     driver.execute(null, "CREATE TABLE testTable(value INTEGER PRIMARY KEY)", 0)
     (0L until 10L).forEach { this.insert(it) }
     transacter = object : TransacterImpl(driver) {}
-  }
-
-  @Test fun `aligned page exhaustion gives correct results`() {
-    val source = KeyedQueryPagingSource(
+    source = KeyedQueryPagingSource(
       queryProvider = this::query,
       pageBoundariesProvider = this::pageBoundaries,
       transacter = transacter,
       context = EmptyCoroutineContext,
     )
-
-    runTest {
-      val expected = (0L until 10L).chunked(2).iterator()
-      var nextKey: Long? = null
-      do {
-        val results = source.load(PagingSourceLoadParamsRefresh(nextKey, 2, false))
-        nextKey = (results as PagingSourceLoadResultPage<Long, Long>).nextKey
-        assertEquals(expected = expected.next(), actual = results.data)
-      } while (nextKey != null)
-    }
   }
 
-  @Test fun `misaligned page exhastion gives correct results`() {
-    val source = KeyedQueryPagingSource(
-      queryProvider = this::query,
-      pageBoundariesProvider = this::pageBoundaries,
-      transacter = transacter,
-      context = EmptyCoroutineContext,
-    )
+  @Test
+  fun aligned_page_exhaustion_gives_correct_results() = runTest {
```

Comment:
My only suggestion would be to wrap `runTest` and `before` together in a helper function so that new tests could be added without needing to remember to call `before()` first.

e.g. https://github.com/cashapp/sqldelight/blob/master/extensions/coroutines-extensions/src/jsTest/kotlin/app/cash/sqldelight/coroutines/RunTest.kt

Reply:
Good idea. I updated the tests to use something like this.